File: backend-go/internal/core/ports/team.go
package ports

import (
	"context"

	"backend-go/internal/core/domain/team"
)

// TeamRepository 战队仓储接口
type TeamRepository interface {
	Create(ctx context.Context, t *team.Team) (*team.Team, error)
	Update(ctx context.Context, id uint, t *team.Team) (*team.Team, error)
	Delete(ctx context.Context, id uint) error
	GetByID(ctx context.Context, id uint) (*team.Team, error)
	List(ctx context.Context, includeInactive bool) ([]team.Team, error)
}

// TeamService 战队服务接口
type TeamService interface {
	CreateTeam(ctx context.Context, req *CreateTeamRequest) (*team.Team, error)
	UpdateTeam(ctx context.Context, id uint, req *UpdateTeamRequest) (*team.Team, error)
	DeleteTeam(ctx context.Context, id uint) error
	GetTeam(ctx context.Context, id uint) (*team.Team, error)
	ListTeams(ctx context.Context, includeInactive bool) ([]team.Team, error)
}

// CreateTeamRequest 创建战队请求
type CreateTeamRequest struct {
	Name      string
	ShortName string
	LogoURL   string
	IsActive  bool
}

// UpdateTeamRequest 更新战队请求
type UpdateTeamRequest struct {
	Name      string
	ShortName string
	LogoURL   string
	IsActive  bool
}

File: src/api/team.ts
import { get, post, put, del } from './http'

export interface Team {
  id: number
  name: string
  shortName?: string
  logoUrl?: string
  isActive: boolean
}

export interface TeamPayload {
  name: string
  shortName?: string
  logoUrl?: string
  isActive: boolean
}

export const getTeams = (includeInactive = false) =>
  get<Team[]>('/api/teams', includeInactive ? { all: 1 } : undefined)

export const createTeam = (payload: TeamPayload) =>
  post<Team>('/api/teams', payload)

export const updateTeam = (id: number, payload: TeamPayload) =>
  put<Team>(`/api/teams/${id}`, payload)

export const deleteTeam = (id: number) => del(`/api/teams/${id}`)

